aws/tests/test_deregister_amis.py

```diff
@@ -1,47 +1,63 @@
 """
 Tests for testeng-ci/aws.
 """
-import boto
-from boto.exception import EC2ResponseError
-from aws.deregister_amis import(
-    main, get_ec2_connection, deregister_amis_by_tag)
-from moto import mock_ec2
-from testfixtures import LogCapture
+from __future__ import absolute_import
+
+import os
 from unittest import TestCase
 
+import boto3
+from botocore.exceptions import ClientError
+from mock import MagicMock, patch
+from testfixtures import LogCapture
+
+from aws.deregister_amis import deregister_amis_by_tag, main
+
+
+class MockImages(object):
+    """
+    Mock boto3 EC2 AMI collection for use in test cases.
+    """
+    def __init__(self, matching_image_exists, filter_raises_error):
+        self.matching_image_exists = matching_image_exists
+        self.filter_raises_error = filter_raises_error
+        self.image = MagicMock()
+        self.image.__str__.return_value = 'test_ami'
+
+    def filter(self, *args, **kwargs):
+        if self.filter_raises_error:
+            raise ClientError(MagicMock(), 'filter')
+        if self.matching_image_exists:
+            return [self.image]
+        return []
+
+
+class MockEC2(object):
+    """
+    Mock boto3 EC2 resource implementation for use in test cases.
+    """
+    def __init__(self, matching_image_exists=True, filter_raises_error=False):
+        self.images = MockImages(matching_image_exists, filter_raises_error)
+
 
-@mock_ec2
 class DeregisterAmisTestCase(TestCase):
     """
     TestCase class for testing get_running_instances.py.
     """
 
     def setUp(self):
-        self.key_id = 'my-key-id'
-        self.secret_key = 'my-secret-key'
-        self.conn = boto.connect_ec2(self.key_id, self.secret_key)
         self.args = [
-            '-i', self.key_id,
-            '-s', self.secret_key,
             '--log-level', 'INFO',
         ]
 
-    def _get_test_image(self):
-        test_image_id = 'ami-11122278'
-        reservation = self.conn.run_instances(test_image_id)
-        self.conn.create_image(
-            name='test-ami',
-            instance_id=reservation.instances[0].id
-        )
-        return self.conn.get_all_images()[0]
-
-    def test_main(self):
+    @patch('boto3.resource', return_value=MockEC2(matching_image_exists=False))
+    def test_main(self, mock_ec2):
         """
         Test output of main
         """
-        with LogCapture() as l:
+        with LogCapture() as capture:
```

Comment:
wow, thank you for changing this variable name! ;)

Reply:
The new pycodestyle complained about it...